### PR TITLE
Deploy on Alpine and add a non-root user

### PR DESCRIPTION
**Pull request by kjsanger**
Fix setting version string in the built application

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,25 +1,34 @@
-FROM golang:1.22
+# syntax = docker/dockerfile:1.2
+
+FROM golang:1.22 as builder
 
 ARG VERSION=dev
```
Comment (jmtcsngr) [inline Dockerfile:5]: Don't think this is being used by make as it is.

**Review by jmtcsngr — CHANGES_REQUESTED** [review]
Using make to call git inside the container feels strange. Passing things through variables may be better. As it is now, it works. So we could create an issue to improve later if you think it is reasonable.

File: Dockerfile (hunk 1 above)
Comment (kjsanger) [reply Dockerfile:5]: Yes, legacy of when I was considering passing all the settings by enviroment variables

**Comment by kjsanger** [comment]
> Using make to call git inside the container feels strange. Passing things through variables may be better. As it is now, it works. So we could create an issue to improve later if you think it is reasonable.

Then we'd have two different build commands, one building in the container and one for outside (the Makefile). We don't worry about copying bash scripts in into containers to run them (it's an accepted way of avoiding hairy `RUN` commands), so I don't think we should worry about Makefiles (especially as `make` is provided by the container).

File: Dockerfile (hunk 1 above)
Comment (jmtcsngr) [reply Dockerfile:5]: OK [PR'ing](https://github.com/wtsi-npg/sqyrrl/pull/19) with line change

**Comment by jmtcsngr** [comment]
I found it unusual to mount `.git` in container build to resolve the version. Also had issues executing tests with `make` so I think there will be more work on the Makefile anyway. I had to check how things were working in actions to figure out how to do things locally.

**Comment by kjsanger** [comment]
The container doesn't install the test dependencies - it's purely for building after the tests have been run by the test workflow. The test framework is installed for the tests here https://github.com/wtsi-npg/sqyrrl/blob/404291adcb3bb86c56a91bcee71e1693de7bda47/.github/workflows/run-tests.yml#L59

**Pull request merged**